services/subscriber/service.go

```diff
@@ -80,55 +136,82 @@ func (s *Service) Open() error {
 		return nil // Service disabled.
 	}
 
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	if s.MetaClient == nil {
-		return errors.New("no meta store")
-	}
-
-	s.closed = false
+	err := func() error {
+		s.mu.Lock()
+		defer s.mu.Unlock()
+		if s.MetaClient == nil {
+			return errors.New("no meta store")
+		}
 
-	s.closing = make(chan struct{})
-	s.update = make(chan struct{})
-	s.points = make(chan *coordinator.WritePointsRequest, 100)
+		s.closing = make(chan struct{})
```

Comment:
Does this line & below need to be in the nested func? Could we move them out for readability?

Reply:
I think it is simpler this way - everything except `updateSubs` and the config check is under the lock, and so protected by the lock. The update to `closing` in particular I think should be under the lock.